fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/ReduceCastRule.java

```diff
@@ -104,4 +117,66 @@ public boolean checkCastTypeReduceAble(Type parent, Type child, Type grandChild)
         Type parentCompatibleType = Type.getAssignmentCompatibleType(child, parent, true);
         return childCompatibleType != Type.INVALID && parentCompatibleType != Type.INVALID;
     }
+
+    public ScalarOperator optimizeDateTimeToDateCastSituation(BinaryPredicateOperator operator) {
+        ScalarOperator child1 = operator.getChild(0);
+        ScalarOperator child2 = operator.getChild(1);
+        if (!(child1 instanceof CastOperator && child2.isConstantRef())) {
+            return operator;
+        }
+
+        ScalarOperator castChild = child1.getChild(0);
+        // optimize cast
+        LocalDateTime originalDateTime = ((ConstantOperator) child2).getDatetime();
+        LocalDateTime bottomLocalDateTime = ((ConstantOperator) child2).getDatetime().toLocalDate().atTime(0, 0, 0, 0);
+        LocalDateTime targetLocalDateTime;
+        BinaryPredicateOperator.BinaryType binaryType = operator.getBinaryType();
+        int offset;
+        BinaryPredicateOperator.BinaryType newBinaryType;
+        if (binaryType.equals(BinaryPredicateOperator.BinaryType.GE)) {
```

Comment:
plz add some comment like example (from old binary predicate to new binary predicate )to help people understand